### DISCUSSION TITLE
HttpServer ThroughStream limitted to 6 simultaneous connections

### DISCUSSION THREAD

**Discussion by koekaverna** [reactphp/reactphp#507] [Q&A] [answered]
I tried to make [SSE ](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)server using **HttpServer** and **ThroughStream**. It works only for 6 connections. First 6 clients can connect and receive streams, but when i connect more clients they can not connect to the server.

I make a reproduction repo for my code: https://github.com/koekaverna/sse-stream

**Comment by netcarver** [comment, marked as answer]
@koekaverna Hello Egor, could you describe how you did the testing? Was each client...

1. a different window/tab in the same browser (say Chrome) on the same client machine? Or
2. in different browsers on the same machine? Or
3. browsers on 6+ different machines, and you could only get 6 of them to connect? Or
4. something else? 

If it's case 1, then you may be running in to the limit that the browser itself places on the number of active SSE connections it allows to the server's domain.  If this is the case, opening a different browser on the same machine should allow you to push beyond 6 connections to your SSE server.

https://stackoverflow.com/questions/18584525/server-sent-events-and-browser-limits

**Reply by koekaverna** [reply]
Hello, @netcarver !
Thank you for the answer.
It was case #1. It works well when I open it in different browsers or using different ips from (127.0.0.1, 127.0.0.2). It's kind of strange behaviour because I open `/sse` path just like regular browser tab (not via JS).
